src/ServiceControl/Persistence/PersistenceFactory.cs

```diff
@@ -1,13 +1,18 @@
 namespace ServiceControl.Persistence
 {
     using System;
+    using System.Collections.Generic;
+    using System.IO;
     using System.Linq;
+    using System.Reflection;
     using System.Runtime.Loader;
     using ServiceBus.Management.Infrastructure.Settings;
     using ServiceControl.Infrastructure;
 
     static class PersistenceFactory
     {
+        static List<string> loadedComponentPersisters = [];
```

Comment:
I'm not sure why this needs to be a list? Isn't a single bool that tracks whether the resolver got added enough? I don't think we need to add it for each type.

Reply:
Wouldn't each component type have its own resolver with different params?

Reply:
Yes I think you are right, although not the path, but the location.  In which case I would use a Hashset instead of a List.  I would also confirm the closure is capturing the correct value.